src/nvim/syntax_defs.h

```diff
@@ -14,13 +14,11 @@ typedef struct syn_state synstate_T;
 #include "nvim/buffer_defs.h"
 #include "nvim/regexp_defs.h"
 
-typedef unsigned short disptick_T;      /* display tick type */
-
 /* struct passed to in_id_list() */
 struct sp_syn {
-  int inc_tag;                  /* ":syn include" unique tag */
-  short id;                     /* highlight group ID of item */
-  short       *cont_in_list;    /* cont.in group IDs, if non-zero */
+  int inc_tag;                  // ":syn include" unique tag
+  int16_t id;                   // highlight group ID of item
+  int16_t *cont_in_list;        // cont.in group IDs, if non-zero
```

Comment:
I guess in theory, padding decisions may differ on some systems based on how "short" is defined. I wonder if explicitly forcing 16 bits will defeat the struct-packing attempt.

It probably doesn't matter too much :)